presentation/valid/ttml2-prstn-shear-linear-mapping.xml

```diff
@@ -13,6 +13,8 @@
     <styling>
       <style xml:id="sRegion" tts:showBackground="whenActive" tts:backgroundColor="black" tts:displayAlign="center"/>
       <style xml:id="sText" tts:color="white" tts:textAlign="center" tts:fontSize="6.67rh" tts:lineHeight="8.0rh"/>
+      <!-- specify tts:bpd on sheared paragraphs, otherwise uses region's bpd (available space) for shear computation -->
+      <style xml:id="sBpd" tts:bpd="16.0rh"/>
```

Comment:
It's not clear to me why the `tts:bpd` attribute has to be added. Can you elaborate? maybe provide a link to the specification section?

Reply:
The current specification text for §10.2.39 `tts:shear` states the following in a note

>This example depicts how a shear (skew) transform is applied to the affected outer block areas generated by p elements; however, when shear is to be applied to an outer block area, then the available distance that can be allocated to the block area, i.e., its maximum inline progression dimension length prior to shear transformation, is reduced by B*sin(θ), where B is the block progression dimension of the outer block area and θ is the shear angle, in order to fit the block area into the content rectangle of the immediate ancestor block area of the paragraph.

Assuming for a moment that B*sin(θ) needs to be corrected to read B*tan(θ), then B is effectively the computed value of `tts:bpd` on the outer block area, i.e., the outer block generated by `p` that serves as a parent for the line areas generated by `p`. Now, the initial value of `tts:bpd` is `auto`, which §10.3.24 defines thus:

>For _bpd_, when applied to an image, the intrinsic size in the block progression direction; otherwise, the numeric value that would be obtained if a value of `100%` were specified.

Now, `100%` is resolved according to §10.3.24:

>If a percentage value is used in a <length> expression of this attribute, it is resolved with respect to the applicable absolute dimension of the generated area's containing area (the nearest ancestor area).

Now, since the containing area of the outer block area of `p` is generated by `div`, and, thence, by `body`, and, thence, by `region`, then we finally resolve the computed value of `tts:bpd='auto'` which, in this case, equals `tts:bpd='100%`, as the height of the horizontal region or the width of the vertical region (in the test).

However, since this computed value exceeds the sum of the _bpd_ of the generated line areas, it ends up over compensating for the skew transform, resulting a reduced effective line measure that doesn't fit (in this case, under fits) the available _ipd_.

In contrast, if `tts:bpd` is specified to be the line height of the two generated line areas, then we get an exact fit.

Reply:
I should also add that an alternative would be `tts:bpd='fitContent'`, however, some implementations may not yet support that, but do support `tts:bpd='<length>'`.